python/ql/src/semmle/python/essa/SsaCompute.qll

```diff
@@ -431,16 +431,40 @@ private module SsaComputeImpl {
       defSourceUseRank(v, b2, 1, i2)
     }
 
+    /**
+     * Holds if `use1` is a use of the variable `v`, and there exists an adjacent reference to `v`
+     * in basic block `b1` at index `i1`.
+     *
+     * A helper predicate for `adjacentUseUseSameVar`, to prevent the first join from being between
+     * the two instances of `variableSourceUse` in
+     * ```ql
+     * exists(SsaSourceVariable v, BasicBlock b1, int i1, BasicBlock b2, int i2 |
+     *     adjacentVarRefs(v, b1, i1, b2, i2) and
+     *     variableSourceUse(v, use1, b1, i1) and
+     *     variableSourceUse(v, use2, b2, i2)
+     * )
+     * ```
+     */
+    pragma[nomagic]
+    cached
```

Comment:
Why `cached` when this is only a helper predicate used by predicate that is already `cached`?

Reply:
Ah, good point. I think what happened here is I started out with a non-`private` helper predicate, which must be cached because the module is. For some reason I decided to make it `cached` rather than `private` (and then later on I made it `private` but forgot to remove the `cached` annotation).

Thanks!

Reply:
Made PR to fix this in https://github.com/github/codeql/pull/4608